spec/acceptance/docker_spec.rb

```diff
@@ -95,6 +97,51 @@ class { 'docker':
       its(:exit_status) { should eq 0 }
       its(:stdout) { should match /docker/ }
     end
+  end
+
+  context 'registry' do
+    before(:all) do
+      registry_host = 'localhost'
+      registry_port = 5000
+      @registry_address = "#{registry_host}:#{registry_port}"
+      @registry_email = 'user@example.com'
+      @manifest = <<-EOS
+        class { 'docker': }
+        docker::run { 'registry':
+          image         => 'registry',
+          pull_on_start => true,
+          ports         => '#{registry_port}:#{registry_port}',
+          volumes       => '/tmp/registry-dev',
+        }
+      EOS
+
+      apply_manifest(@manifest, :catch_failures=>true)
+    end
+
+    it 'should be able to login to the registry' do
+      manifest = <<-EOS
+        docker::registry { '#{@registry_address}':
+          username => 'user',
+          password => 'password',
+          email    => '#{@registry_email}',
+        }
+      EOS
+      apply_manifest(manifest, :catch_failures=>true)
+      shell("cat ~/.docker/config.json | grep #{@registry_address}", :acceptable_exit_codes => [0])
```

Comment:
`grep PATTERN FILENAME` is usually preferable to `cat | grep`, also quoting the pattern would be good practice, although I can see it's not required here.